processing/src/main/java/org/apache/druid/query/groupby/having/HavingSpecMetricComparator.java

```diff
@@ -73,6 +73,9 @@ static int compare(String aggregationName, Number value, Map<String, AggregatorF
           double d = Double.parseDouble(metricValueStr);
           return Double.compare(d, value.doubleValue());
         }
+      } else if (metricValueObj instanceof Number) {
+        BigDecimal d = new BigDecimal(metricValueObj.toString());
+        return d.compareTo((BigDecimal) value);
```

Comment:
Maybe we need to create another BigDecimal from value as well rather than casting?

Reply:
This code assumes that Numbers that are not Float/Double/Integer/Long are BigDecimal, which is not correct.